src/rnpkeys/rnpkeys.cpp

```diff
@@ -225,24 +219,56 @@ import_keys(cli_rnp_t *rnp, rnp_input_t input, const std::string &inname)
             json_object_put(jso);
             break;
         }
+        processed_keys += json_object_array_length(keys);
         for (size_t idx = 0; idx < (size_t) json_object_array_length(keys); idx++) {
             json_object *    keyinfo = json_object_array_get_idx(keys, idx);
             rnp_key_handle_t key = NULL;
-            if (!keyinfo || !imported_key_changed(keyinfo)) {
+            if (!keyinfo) {
                 continue;
             }
+            std::string pub_status = json_obj_get_str(keyinfo, "public");
+            std::string sec_status = json_obj_get_str(keyinfo, "secret");
             const char *fphex = json_obj_get_str(keyinfo, "fingerprint");
+
+            if (pub_status == "new") {
+                new_pub_keys.insert(fphex);
+                updated = true;
+            }
+            if (sec_status == "new") {
+                new_sec_keys.insert(fphex);
+                updated = true;
+            }
+            if (pub_status == "updated" || sec_status == "updated") {
+                updated_keys.insert(fphex);
+                updated = true;
+            }
+            if (pub_status == "unchanged" || sec_status == "unchanged") {
+                if (!new_pub_keys.count(fphex) && !new_sec_keys.count(fphex) &&
+                    !updated_keys.count(fphex)) {
+                    unchanged_keys++;
+                    continue;
+                }
+            }
             if (rnp_locate_key(rnp->ffi, "fingerprint", fphex, &key) || !key) {
                 ERR_MSG("failed to locate key with fingerprint %s", fphex);
                 continue;
             }
             cli_rnp_print_key_info(stdout, rnp->ffi, key, true, false);
             rnp_key_handle_destroy(key);
-            updated = true;
         }
         json_object_put(jso);
     } while (1);
 
+    // print statistics
+    ERR_MSG("Import finished: %lu key%s processed, %lu new public keys, %lu new secret keys, "
```

Comment:
As from the user's point of view it would be preferred (and more aligned with other messages) to not display data about the 0 new/updated keys. I.e. something like the following:
- Import finished: 3 keys processed, 1 new public key, 1 updated and 1 unchanged.
- Import finished: 3 keys processed, 2 updated, 1 unchanged.
- Import finished: 3 keys processed, 3 unchanged. 

Looks a bit complicated from the logic and tests perspective, but should be doable.

Reply:
Yes, but it would be inconvenient for users who want to use `rnpkeys` output in some bash pipelines or scripts or run it in a background from the GUI wrapper (for example).

And we already have the similar message for imported signatures which could have zero values
https://github.com/rnpgp/rnp/blob/d1915c1e8a1d3ac5acc562c777cfcd56e79942df/src/rnpkeys/rnpkeys.cpp#L302

Reply:
@ni4 any thoughts on this? Regarding command outputs, I wonder if we can do some JSON/YAML type of output for people who want to use `rnpkeys` as a simple API.

Reply:
@ronaldtse This output should be considered as 'for human eyes only' (and for out CI scripts :) ). In case of automated processing we could add some CLI params. Another possible solution for automated processing is to use our FFI via Bash scripts (didn't try that yet but looks promising) :  https://github.com/taviso/ctypes.sh

Reply:
This looks pretty hard to integrate for achieving JSON based output of FFI functions?

Reply:
> This looks pretty hard to integrate for achieving JSON based output of FFI functions?

We'll just need to consolidate outputs of the `rnp_import_keys()`, i.e. return fingerprints of imported keys, with attached info whether key was updated/new or already existing.

Reply:
...okay, let's merge it now. If we'll need to change the output to be less redundant, we may do that via the separate PR.